packages/puppeteer-core/src/bidi/Frame.ts

```diff
@@ -360,8 +362,34 @@ export class BidiFrame extends Frame {
                   })
                 )
               ),
-              map(() => {
-                return navigation;
+              switchMap(() => {
+                if (navigation.request) {
+                  function requestFinished$(
+                    request: Request
+                  ): Observable<Navigation> {
+                    if (request.response || request.error) {
+                      return of(navigation);
+                    }
+                    if (request.redirect) {
+                      return requestFinished$(request.redirect);
+                    }
+                    // Reduces flakiness if the response events arrive after
+                    // the load event.
+                    // Usually, the response or error is already there at this point.
```

Comment:
I think this comment is better suited for the `if (request.response || request.error) {` as here it implies that the `success`,`error` events most of the time are not emitted.